### ISSUE TITLE
New Logitech G915 battery not detected/unknown

### ISSUE THREAD

**Issue by Sphiinx** [pwr-Solaar/Solaar#596]
**Information**
Add support for the newly released Logitech G915 Keyboard. It's connected via the lightspeed wireless dongle.

UPower and Solaar show nothing.

https://www.logitechg.com/en-us/products/gaming-keyboards/g915-low-profile-wireless-mechanical-gaming-keyboard.html

**Comment by FFY00** [comment]
I am pretty sure it uses the `0x1001` feature but can you paste the output of `solaar show`.

**Comment by Phernost** [comment]
I made bare minimal changes to add `c541` to the udev rules, `base_usb.py`, and `receiver.py` to make it show up right

```
Unifying Receiver
  Device path  : /dev/hidraw5
  USB id       : 046d:c541
  Serial       : 60E94F4B
    Firmware   : 01.01.B0027
    Bootloader : 00.20
    Other      : B5.53
  Has 1 paired device(s) out of a maximum of 1.
  Notifications: wireless, software present (0x000900)
  Device activity counters: (empty)

  1: G915 WIRELESS RGB MECHANICAL GAMING KEYBOARD
     Codename     : G915 KEYBOARD
     Kind         : keyboard
     Wireless PID : 407C
     Protocol     : HID++ 4.2
     Polling rate : 8 ms (125Hz)
     Serial number: CA5A32C2
        Bootloader: BOT 77.00.B0031
             Other: 
          Firmware: MPK 09.01.B0035
             Other: 
             Other: 
     The power switch is located on the top left corner.
     Supports 39 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
         3: DEVICE NAME            {0005}   
         4: WIRELESS DEVICE STATUS {1D4B}   
         5: RESET                  {0020}   
         6: unknown:0007           {0007}   
         7: unknown:1001           {1001}   
         8: CHANGE HOST            {1814}   
         9: unknown:1815           {1815}   
        10: unknown:8071           {8071}   
        11: unknown:8081           {8081}   
        12: REPROG CONTROLS V4     {1B04}   
        13: unknown:1BC0           {1BC0}   
        14: ENCRYPTION             {4100}   
        15: unknown:4522           {4522}   
        16: KEYBOARD LAYOUT 2      {4540}   
        17: GKEY                   {8010}   
        18: MKEYS                  {8020}   
        19: MR                     {8030}   
        20: unknown:8040           {8040}   
        21: ONBOARD PROFILES       {8100}   
        22: REPORT RATE            {8060}   
        23: unknown:00C2           {00C2}   
        24: DFU                    {00D0}   
        25: unknown:1802           {1802}   internal, hidden
        26: unknown:1803           {1803}   internal, hidden
        27: unknown:1806           {1806}   internal, hidden
        28: unknown:1813           {1813}   internal, hidden
        29: unknown:1805           {1805}   internal, hidden
        30: unknown:1830           {1830}   internal, hidden
        31: unknown:1890           {1890}   internal, hidden
        32: unknown:1891           {1891}   internal, hidden
        33: unknown:18A1           {18A1}   internal, hidden
        34: unknown:1DF3           {1DF3}   internal, hidden
        35: unknown:1E00           {1E00}   hidden
        36: unknown:1EB0           {1EB0}   internal, hidden
        37: unknown:1861           {1861}   internal, hidden
        38: unknown:18B0           {18B0}   internal, hidden
     Has 2 reprogrammable keys:
         0: unknown:00D1              , default: unknown:00AE                => unknown:00D1              
             divertable, persistently divertable, pos:1, group:0, gmask:0
         1: unknown:00D2              , default: unknown:00AF                => unknown:00D2              
             divertable, persistently divertable, pos:2, group:0, gmask:0
     Battery status unavailable.

```

**Comment by FFY00** [comment]
Confirmed, it uses the `0x1001` feature.

**Comment by pfps** [comment]
So this should be fixed by #632, right?

**Comment by pfps** [comment]
Closing due to no response from original poster.
